Deployment

mikethreels opened the pull request:
## Milestone 4: Deployment

### Summary
- [x] implemented model for friendship
- [x] implemented friendship features like friend requests and accepting/rejecting friend request
- [x] display mutual friends on the users profile page
- [x] displaying post from user and friends on the timeline
- [x] implemented integration and unit tests

mikethreels committed:
created the ERD

santiagorodriguezbermudez committed:
Updated linter documentation

santiagorodriguezbermudez committed:
Updated rubocop and styelint files

mikethreels committed:
Merge pull request #1 from mikethreels/feature-erd

created the ERD

santiagorodriguezbermudez committed:
Added friendship associations

mikethreels committed:
added friendship arrray

mikethreels committed:
added invite method

santiagorodriguezbermudez committed:
Added invite button to every profile

santiagorodriguezbermudez committed:
Accept or reject invitations

santiagorodriguezbermudez committed:
Clear out views from index and invitations

mikethreels committed:
filtered out the post of none friends

mikethreels committed:
fixed rubocop errors

mikethreels committed:
fixed sign-up error, created friendship controller

santiagorodriguezbermudez committed:
Added logic from the view to helpers

santiagorodriguezbermudez committed:
Added unit testing to the app

santiagorodriguezbermudez committed:
Fixed rubo offenses

santiagorodriguezbermudez committed:
Added integration tests to the user flow

mikethreels committed:
adjusted the readme file

mikethreels committed:
Merge pull request #2 from mikethreels/feature-friendship-v1

Feature friendship v1

mikethreels committed:
added all friend list to show user

mikethreels committed:
implemented mutual friend list on user profile and fixed rubocop/linter errors

santiagorodriguezbermudez committed:
Moved logic to the model

mikethreels committed:
added integration tests

mikethreels committed:
fixed rubocop errors

mikethreels committed:
Merge pull request #3 from mikethreels/feature-friendship-v2

Feature friendship v2

stratospherique reviewed (changes requested):
### Great work @team :clap: :clap: However, I noticed some issues that you need need to fix them in order   to get approval for this project. Check :arrow_down: 

- **[REQUIRED]** The deployed website should reflect the latest version of your code. Kindly push your latest changes to production.
- **[REQUIREd]** some issues from previous milestones have surfaced, :pray: fix them
* [x] user should be able to see posts of another user unless he becomes friends with
* [x] user shouldn't be able to send an invitation to a user that they have a pending invitation with

### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

santiagorodriguezbermudez committed:
Fixed friend request

mikethreels committed:
Merge pull request #5 from mikethreels/feature-repair

Fixed friend request

mikethreels commented:
## Change log:
- [x] removed the possibility to invite a person who already invited the current user
- [x] implemented above change in both the index page and the profile page
- [x] made sure posts of other users who are not friends are not visible in the timeline.
- [x] deployed the develop branch to Heroku

mcrd25 reviewed (approved):
# Changes Approved :heavy_check_mark: 

Hi Team! :wave: 

All seems good now! You may merge and deploy!


***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)
